subiquity/server/autoinstall.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2024 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import logging
+
+from jsonschema.exceptions import ValidationError
+
+log = logging.getLogger("subiquity.server.autoinstall")
+
+
+class AutoinstallError(Exception):
```

Comment:
Do we need to build out this functionality?  The result is that AutoinstallError doesn't really behave like other exceptions - str() and repr() look very different, we can get to the message on an exception with str(exception), etc.  Basically, what are the advantage of this approach over an declaration like:
```
class AutoinstallError(Exception):
    pass
```

Reply:
You're right. I don't think any of the logic here in `AutoinstallError` is necessary. Mostly remnant of old ideas. Any special handling should be done by the subclass (like `__str__` in `AutoinstallValidationError).